django_loci/static/django-loci/js/loci.js

```diff
@@ -198,6 +199,15 @@ django.jQuery(function ($) {
       $geoEdit.hide();
       $indoor.hide();
       $typeRow.show();
+    }
+      // Show alert when switching from indoor to outdoor if floorplans exist
+    if (value === "outdoor" && floorplans_length > 1){
+      var msg =
+        "Please remove the associated floorplans first " +
+        'and save; then you can switch to type "indoor"';
+      alert(gettext(msg));
```

Comment:
I have an idea which may be better. What if instead of using an alert and ask the user to do this action, we do it for them after asking for confirmation with a confirm dialog? If they hit yes we remove the floorplan and hit the save with JS. Better UX. What do you think @niteshsinha17?

Reply:
@nemesifier Please confirm, This also means if the user has changed the device name and trying to change the type and hits yes this will lead to saving changes of device name in the db. If yes, then I think it's not a better UX.

I think this can be a better option,

Disable editing of type and show some help text to remove floor plans first to change the type. This can be done via both JS and Python code.

What do you think?

Reply:
Also, in inline forms, js is not running, 

This js code has been added to show an alert when a user tries to change the type but I can't find its log in the console.

https://github.com/openwisp/django-loci/blob/master/django_loci/static/django-loci/js/floorplan-inlines.js

Reply:
@niteshsinha17 yes in the current code floorplan-inlines.js is used only for location form.

Reply:
Should we do this for normal location form too?

Reply:
@nemesifier  I have added a commit to reflect this. For normal location form, i dont have access to floorplan instance in location form so i handled it via LocationAdmin to disable its validation/creation when location is outdoor. I have kept this handling mainly on python code and removed js code for location change alert and any other previous code which was added by me in js. Also updated the test cases for the same